core/src/view/view.cpp

```diff
@@ -134,7 +135,12 @@ void View::setRoll(float _roll) {
 
 void View::setPitch(float _pitch) {
 
-    m_pitch = glm::clamp(_pitch, 0.f, (float)HALF_PI);
+    float max_pitch = HALF_PI;
+    if (m_type != CameraType::perspective) {
+        // Prevent projection plane from intersecting ground plane
+        max_pitch = atan2(m_pos.z, m_height * .5f);
```

Comment:
Checking the max won't fix this in the general case: Consider resizing a portrait-oriented window by hand and making it taller with the same width. Height is still the larger dimension but you've just created the same problem with pitch.